src/throttled-transform.js

```diff
@@ -1,188 +0,0 @@
-/* eslint-disable no-underscore-dangle */
-// TODO: use NPM package
-import ParallelTransform from '../../node-parallel-transform-stream/dist/parallel-transform'; // eslint-disable-line max-len
-
-const _qps = new WeakMap(),
-  _bucketRunning = new WeakMap(),
-  _queries = new WeakMap(),
-  _queue = new WeakMap();
-
-export default class ThrottledTransform extends ParallelTransform {
-  /**
-   * ThrottledTransform instance
-   * All child class must implement the `_throttledTransform` function.
-   * Child classes should not implement the `_transform` and `_flush` functions.
-   *
-   * @param {Object} options Options which will be passed to the
-   *                         `stream.Transform` constructor
-   **/
-  constructor(options = {}) {
-    const queriesPerSecond = options.queriesPerSecond || 35,
-      defaultOptions = {
-        maxParallel: queriesPerSecond
-      };
-
-    super(Object.assign({}, defaultOptions, options));
-
-    _qps.set(this, queriesPerSecond);
-    _bucketRunning.set(this, false);
-    _queries.set(this, 0);
-    _queue.set(this, []);
-  }
-
-  /**
-   * Helper function for easily creating ThrottledTransform streams
-   *
-   * @param {Function} transform The stream's _transform function
-   * @param {Function} flush The stream's _flush function
-   * @param {Object} defaultOptions Default options for the class constructor
-   * @returns {Class} A ParallelTransform class
-   **/
-  static create(transform, flush = done => done(), defaultOptions = {}) {
-    class Transform extends ThrottledTransform {
-      constructor(options) {
-        super(Object.assign({}, defaultOptions, options));
-      }
-
-      _throttledTransform = transform;
-      _throttledFlush = flush;
-    }
-
-    return Transform;
-  }
-
-  /**
-   * Executed every time data was pushed into the stream
-   * @param {?}        data Data to be transformed
-   * @param {string}   encoding Encoding, if `chunk` is a string
-   * @param {Function} done Callback which must be executed
-   *                        when transformations have finished
-   **/
```

Comment:
Hmm, `query` is using the rest-operator, so the function technically accepts 3 arguments, not just "query".